task/task.ps1

```diff
@@ -44,7 +44,7 @@ try
     {
         throw "Provided configuration file path $ssrsFilePath is not valid."
     }
-
+    Clear-SsrsFolderItems -Folder $folder -Proxy $proxy
```

Comment:
Does this really need to be called always? Meanwhile traversing folders, can't we verify if the clear for the folder is set, then perform the clean?

Reply:
The script will clean each folder based on if the CleanExistingItems value is set to true for the folder. 
Clear-SsrsFolderItems is called recursively over the folder tree defined in the config. Due to this, the CleanExistingItems check is done inside the function call so it may be executed for each directory in the tree.
This is useful if you need to clean a nested folder but want to preserve its parent directory.

Reply:
I was thinking more of adding the necessary into the Publish-SsrsFolder cmdlet, it is already traversing all of the folders, and there, if the flag is on, do the cleanup.

Reply:
Ah, OK. I see what you mean. My original idea was to also remove any unused folders by starting at the leaf nodes and pruning them as the script deletes the files. Decided against it in the end for the sake of simplicity but thats why it is traversing the tree twice. 
It makes more sense to only do it once, I'll rework the code to clean the files in the publish cmdlet.

Reply:
Updated to do the clean within the publish folder cmdlet